src/averageAges.js

```diff
@@ -55,7 +67,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(child =>
+      people.some(mother => child.mother === mother.name)
+      && child.sex === 'm')
+    : people.filter(child =>
+      people.some(mother => child.mother === mother.name));
```

Comment:
```suggestion
    : people.filter(child => (
      people.some(mother => child.mother === mother.name)
    );
```
it will be more readable approach